src/main/java/io/cryostat/agent/ConfigModule.java

```diff
@@ -251,12 +266,75 @@ public static int provideCryostatAgentWebclientResponseTimeoutMs(Config config)
         return config.getValue(CRYOSTAT_AGENT_WEBCLIENT_RESPONSE_TIMEOUT_MS, int.class);
     }
 
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PATH)
+    public static Optional<String> provideCryostatAgentWebclientTlsTruststorePath(Config config) {
+        return config.getOptionalValue(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PATH, String.class);
+    }
+
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_FILE)
+    public static Optional<String> provideCryostatAgentWebclientTlsTruststorePassFromFile(
+            Config config,
+            @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_CHARSET) String passCharset) {
+        Optional<String> truststorePassFile =
+                config.getOptionalValue(
+                        CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_FILE, String.class);
+        Optional<String> password = Optional.empty();
+        try (FileInputStream passFile = new FileInputStream(truststorePassFile.get())) {
+            String pass = IOUtils.toString(passFile, Charset.forName(passCharset));
+            pass = pass.substring(0, pass.length() - 1);
+            password = Optional.ofNullable(pass);
+        } catch (NoSuchElementException e) {
+            return password;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return password;
```

Comment:
We should be careful with the passwords here. I think this is going to result in the password (or passfile content) getting left with a dangling reference in the dependency injection system, which will leave the password contents allocated on the heap forever.

It would be better if this was something like `Optional<ByteBuffer>`, where

```java
public class ByteBuffer {
  private final byte[] buf;
  public ByteBuffer(int len) {
    this.buf = new byte[len];
  }
  public void set(String s) {
    assert s.length() <= this.buf.length;
    Arrays.copy(s.getBytes(), this.buf);
  }
  public String get() {
    return new String(this.buf);
  }
  public void clear() {
    Arrays.fill(this.buf, '\0');
  }
```

(rough pseudojava)

This way `ByteBuffer#clear()` can be called after we're done initializing the truststore and the password is only present on the heap for the shortest amount of time possible. It will still be present somewhere in the environment, as an environment variable or system property etc., but we can minimize the surface area.